woocommerce/woocommerce-gateway-stripe
# Show UPE payment methods in saved order on block checkout page

Fixes #2994 

In https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3184 we have implemented the ability to order the payment methods on the Stripe settings page when UPE is enabled but the order does not reflect on the block checkout. In this PR, I have fixed that and the order of the UPE methods on block checkout is now the same as Stripe settings page.

## Changes proposed in this Pull Request:
- In [blocks code](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Blocks/Payments/Api.php#L86), the payment methods are retrieved from `WC()->payment_gateways->payment_gateways()` (all gateways). For that, the `woocommerce_available_payment_gateways` filter does not work on the block checkout for reordering the payment methods.
- I am now updating the `woocommerce_gateway_order` option which holds all payment gateway order when the Stripe payment methods order is changed. In `add_stripe_methods_in_woocommerce_gateway_order` we find the main Stripe gateway position in `woocommerce_gateway_order` and push all the other Stripe methods there in the order saved in the Stripe settings page. 
- `WC()->payment_gateways->payment_gateways()` returns the list based on the order of `woocommerce_gateway_order`, so now the order of payment methods on blocks checkout reflects the order in the Stripe settings page.

## Testing instructions
- Enable the new checkout experience.
- On the Stripe settings page, confirm that there is a `Change display order` button at the top of the payment method list.
- Change the order of a few methods and save changes.
- As a shopper, add product to your cart and go to the block checkout page. Confirm that the Stripe methods are displayed in the saved order.
- Go to the shortcode checkout page and confirm that the order is correct here as well.
- In your admin dashboard, go to **WooCommerce > Settings > Payments**. Change the order of a few gateways and enable one or two gateways along with Stripe.
- Check the shortcode and block checkout pages and confirm that the payment method order is correct.
